code/game/objects/effects/particle_holder.dm

```diff
@@ -12,24 +12,29 @@
 	/// See \code\__DEFINES\particles.dm
 	var/particle_flags = NONE
 
-	var/atom/parent
+	var/datum/weakref/parent_ref
```

Comment:
There's no point in changing this to use a weakref, we already manage the reference with the qdel signal

Someone just forgot to put `parent = null` in `Destroy`